src/inmanta/deploy/scheduler.py

```diff
@@ -673,15 +678,22 @@ async def report_resource_state(
                     state.deployment_result = deployment_result
                 return
 
-            if skipped_for_dependency:
-                state.blocked = BlockedStatus.TRANSIENT
             # We are not stale
             state.status = status
             if deployment_result is not None:
                 # first update state, then send out events
                 self._deploying_latest.remove(resource)
                 state.deployment_result = deployment_result
                 self._work.finished_deploy(resource)
+                # Check if we need to mark a resource as transiently blocked
+                # Or if we can unblock a transiently blocked resource (its dependencies now succeed)
+                if skipped_for_dependency and state.blocked is not BlockedStatus.YES:
+                    dependencies = await self._get_last_non_deploying_state_for_dependencies(resource=resource)
+                    state.blocked = (
+                        BlockedStatus.NO
+                        if all(state is const.ResourceState.deployed for state in dependencies.values())
+                        else BlockedStatus.TRANSIENT
+                    )
                 if deployment_result is DeploymentResult.DEPLOYED:
```

Comment:
If it does deploy successfully, we may want to unblock dependents, as mentioned in a previous comment. From the absence of that logic, do I understand correctly that you've passed on this part to Hugo's PR?

Reply:
I think I can do it on my side easily.

Reply:
I underestimated this. I'll talk with @Hugo-Inmanta tomorrow morning

Reply:
Let me know if you want me to join in.